nomad/resource_variable.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	// maxPathLength is defined by Nomad here:
+	// https://github.com/hashicorp/nomad/blob/c7c8bd4ce2fcd9f2cd13278043b7baf1f8dd4f47/nomad/structs/variables.go#LL158
+	maxPathLength = 128
+)
+
+var (
+	validVariablePath = regexp.MustCompile("^[a-zA-Z0-9-_~/]+$")
+)
+
+func resourceVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVariableWrite,
+		Update: resourceVariableWrite,
+		Delete: resourceVariableDelete,
+		Read:   resourceVariableRead,
+		Exists: resourceVariableExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Description:  "The path at which the variable items will be stored, must be between 1 and 128 characters in length, be URL safe, and not include '@' or '.' characters",
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
```suggestion
				Required:     true,
				ForceNew:     true,
```

I think updating the variable path requires a destructive update, otherwise we risk leaving stray variables when the path is changed.